client/src/components/TripElement/NewTripForm.js

```diff
@@ -1,28 +1,34 @@
 import { useDispatch, useSelector } from 'react-redux'
 import { AppView } from '../../constants/enums'
 import { closeNewTripModal } from '../../redux/reducers/modalsSlice'
-import { addTrip, selectCurrentId } from '../../redux/reducers/userSlice'
+import { createTripAsync } from '../../redux/reducers/trip/thunks'
 import { setActiveTripId, setAppView } from '../../redux/reducers/viewSlice'
 import { Modal } from '../common'
 import { TripForm } from './TripForm'
 
 export function NewTripForm() {
+  const dispatch = useDispatch()
   const appView = useSelector((state) => state.view.appView)
-  const currentId = useSelector(selectCurrentId)
+  const trips = useSelector((state) => state.trip.trips)
+
   const newTripModalIsOpen = useSelector(
     (state) => state.modals.newTripModalIsOpen,
   )
-  const dispatch = useDispatch()
 
   const handleCloseNewTripModal = () => {
     dispatch(closeNewTripModal())
   }
 
   const onSubmit = (data) => {
-    // TODO: make api call to create new trip
-    console.log(data)
-    dispatch(addTrip(data))
-    dispatch(setActiveTripId(currentId))
+    // this is default tripName, lat, long, TODO: get from other endpoints
+    const tripDataWithTripName = {
+      ...data,
+      tripName: `Your Trip ${trips.length + 1}`,
+      destinationLatitude: 49.23990319450836,
+      destinationLongitude: -123.15644121337681,
+    }
+    dispatch(createTripAsync(tripDataWithTripName))
+    dispatch(setActiveTripId(trips[trips.length - 1]._id))
```

Comment:
nit: do you think theres any way to get the id from response to use here? using last element in array seems hacky.